webpack/assets/javascripts/hosts/tableCheckboxes.js

```diff
@@ -68,6 +73,15 @@ function toggleActions() {
 // setups checkbox values upon document load
 $(document).on('ContentLoad', () => {
   if (window.location.pathname !== foremanUrl('/hosts')) return;
+  const hostQuery = sessionStorage.getItem('hostQuery');
+  const uriSearch = getURIsearch();
+  if (hostQuery && hostQuery !== uriSearch) {
+    cleanHostsSelection();
+    sessionStorage.setItem('hostQuery', uriSearch);
```

Comment:
isn't pagination is saved across page views? (so you can always take it from the URL?) it would be a major bug if it doesn't...?